Реализован функционал 10 Спринта

avralevm opened the pull request:
1)Созданы 2 класса POJO
2)Созданы классы-контроллеры
3)Создан класс-валидатор 
4)Добавлено логирование операций, изменяющих сущность
5)Добавлены тесты для проверки функционала

avralevm committed:
Коммит

avralevm committed:
Исправлен тест validateEmailTest_EmptyEmail

avralevm commented:
Привет, валидацию разбил на простую, которая проверяется аннотациями и сложную в класс Validators, стоит ли методы валидации перенести в Controler или в сам класс POJO. 
Так же посмотрел на stackoverflow, что можно свои аннотации создавать, стоит ли методы валидации переделывать в аннотации или это пожеланию? 
Так же немного не нравится, что сервисная логика вся находится в контроллерах, хотя вроде в ТЗ говорилось, что это нормально. Стоит сейчас переделать реализацию или оставить на следующие спринты?

litvitnik commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,52 @@
+    Map<Long, Film> films = new HashMap<>();
+    private long countID = 0;
+
+    private long getNextID() {
```
Приватные методы должны быть в конце класса

litvitnik commented:
Привет! Можно оставить проверку условия, согласно которому имя для отображения может быть пустым (в таком случае будет использоваться логин), в классе Validators. А вот проверку для условия "дата релиза не должна быть раньше 28 декабря 1895 года" лучше сделать аннотацией, то есть написать собственную аннотацию, которая будет проверять это условие. Что касается сервисной логики, в следующем спринте планируется переработка архитектуры, и вся логика будет перемещена в сервисы. Поэтому на данный момент вносить изменения не нужно, в этом спринте вся логика должна оставаться в контроллерах)

avralevm committed:
1)Добавлено 2 кастомные аннотации
2)В классы POJO добавлены созданные аннотации
3)Добавлено логирование Exception
4)Переработан ответ сервера на Exception
5)Скорректированы тесты

avralevm committed:
Изменил аннотацию, объединил с @NotBlank

avralevm committed:
Commit
